aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -599,6 +599,109 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
   }
 }
 
+void pre_check_gradient(const Tensor& self, IntArrayRef axis, int64_t spacing_size,  int64_t edge_order) {
+  TORCH_CHECK(self.scalar_type() != ScalarType::Byte, "torch.gradient does not support uint8 as input.");
+  TORCH_CHECK( spacing_size == axis.size(), "torch.gradient should receive axis and spacing size to be equal.");
+  TORCH_CHECK( edge_order == 1, "torch.gradient supports edge_order value only to be equal to 1.");
+  dim_list_to_bitset(axis, self.dim());
+}
+
+std::vector<Tensor> gradient_helper(const Tensor& self, TensorList spacing, IntArrayRef axis, int64_t edge_order) {
+  for (const auto i : c10::irange(spacing.size())) {
+    TORCH_CHECK(self.device() == spacing[i].device(), "torch.gradient requires each tensor in spacing to be in the same device as self.");
```

Comment:
"torch.gradient requires each tensor in spacing to be in the same device as self." -> "torch.gradient expected each tensor to be on the same device, but got devices [self.device] and [space[i].device()]!"